src/Mvc/Mvc.Core/src/Builder/OpenApiEndpointConventionBuilderExtensions.cs

```diff
@@ -0,0 +1,103 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Routing;
+
+namespace Microsoft.AspNetCore.Http
+{
+    /// <summary>
+    /// Extension methods for adding response type metadata to endpoints.
+    /// </summary>
+    public static class OpenApiEndpointConventionBuilderExtensions
+    {
+        private static readonly SuppressApiMetadata _suppressApiMetadata = new();
+        /// <summary>
+        /// Adds metadata to support suppressing OpenAPI documentation from
+        /// being generated for this endpoint.
+        /// </summary>
+        /// <param name="builder">The <see cref="IEndpointConventionBuilder"/>.</param>
+        /// <returns>A <see cref="IEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+        public static IEndpointConventionBuilder SuppressApi(this IEndpointConventionBuilder builder)
```

Comment:
```suggestion
        public static TBuilder SuppressApi<TBuilder>(this TBuilderbuilder) where TBuilder : IEndpointConventionBuilder
```
Why not use the same pattern like other methods for adding metadata?